framework/test_selectors/SelectorFactory.cpp

```diff
@@ -195,10 +197,17 @@ void generate_test_list(std::vector<struct test *> &test_list, std::span<struct
         if (!SandstoneConfig::RestrictedCommandLine && sApp->fatal_skips)
             fprintf(stderr, "# WARNING: --fatal-skips used with full test suite. This will probably fail.\n"
                             "# You may want to specify a controlled list of tests to run.\n");
-        /* generate test list based on quality levels only */
-        for (struct test &test : test_set) {
-            if (test.quality_level >= min_quality)
-                add_test(test_list, &test);
+        auto builtin_test_list = get_test_list("auto");
+        if (builtin_test_list) {
+            for (auto* test : *builtin_test_list) {
+                if (test->quality_level >= min_quality)
+                    add_test(test_list, test);
+            }
+        } else {
+            for (auto& test : test_set) {
+                if (test.quality_level >= min_quality)
+                    add_test(test_list, &test);
+            }
```

Comment:
I don't think this is the right place to put this. I would prefer for it to be in sandstone.cpp's `main()`, where we're already calling `get_test_lsit()` (actually, there may be no need for modifications, at all, if you make the function handle a `nullptr` name be the same as "auto").